Packs/IvantiHeat/Integrations/IvantiHeat/IvantiHeat.py

```diff
@@ -0,0 +1,290 @@
+    if limit:
+        params['$top'] = limit
+    if offset:
+        params['$skip'] = offset
+    if query:
+        params['$search'] = query
+
+    raw_res = client.do_request('GET', f'odata/businessobject/{object_type}', params=params)
+    if raw_res.get('value'):
+        human_readable = tableToMarkdown(f'{object_type} results', t=raw_res.get('value'), headers=OBJECT_HEADERS,
+                                         removeNull=True)
+
+        entry_context = {f'IvantiHeat.{object_type}(val.RecId && val.RecId===obj.RecId)': raw_res.get('value')}
+        return human_readable, entry_context, raw_res
+
+    return 'No records found', {}, raw_res
+
+
+@logger
+def update_object_command(client, args):
+    object_type = args.get('object-type')
+    rec_id = args.get('rec-id')
+    fields = args.get('fields')
+
+    try:
+        fields = json.loads(fields)
+    except Exception:
+        raise Exception(f'Failed to parse fields as JSON data, received object:\n{fields}')
+
+    raw_res = client.do_request('PUT', f'odata/businessobject/{object_type}(\'{rec_id}\')', json_data=fields)
+
+    human_readable = tableToMarkdown(f'{rec_id} updated successfully', t=raw_res, headers=OBJECT_HEADERS,
+                                     removeNull=True)
+    entry_context = {f'IvantiHeat.{object_type}(val.RecId===obj.RecId)': raw_res}
+    return human_readable, entry_context, raw_res
+
+
+@logger
+def delete_object_command(client, args):
+    object_type = args.get('object-type')
+    rec_id = args.get('rec-id')
+    raw_res = client.do_request('DELETE', f'odata/businessobject/{object_type}(\'{rec_id}\')')
+    return f'Record {rec_id} deleted successfully', {}, raw_res
+
+
+@logger
+def get_attachment_command(client, args):
+    attachment_id = args.get('attachment-id')
+    raw_res = client.do_request('Get', f'rest/Attachment?ID={attachment_id}', resp_type='other')
+
+    filename_header = raw_res.headers.get('content-disposition')
+
+    f_attr = 'filename='
+    if filename_header and f_attr in filename_header:
+        filename = filename_header[filename_header.index(f_attr) + len(f_attr):]
+        return fileResult(filename, raw_res.content)
+
+
+@logger
+def upload_attachment_command(client, args):
+    object_type = args.get('object-type')
+    entry_id = args.get('entry-id')
+    rec_id = args.get('rec-id')
+
+    body = {'ObjectID': rec_id, 'ObjectType': f'{object_type}#'}
+    raw_res = client.do_request('POST', 'rest/Attachment', data=body, files={'file': get_file(entry_id)})
+    if raw_res:
+        attachment = raw_res[0]
+        attachment_id = attachment.get('Message')
+        file_name = attachment.get('FileName')
+        entry_context = {'IvantiHeat.Attachment':
+                         {'RecId': rec_id, 'AttachmentId': attachment_id, 'FileName': file_name}}
+        return f'{file_name} uploaded successfully, attachment ID: {attachment_id}', entry_context, raw_res
+    raise Exception(f'Upload attachment {rec_id} failed')
```

Comment:
will the response have information about the failure reason? if so, need to add it to the exception message

Reply:
when the request failed, the exacption catch and the failure information returns to the war room:
![image](https://user-images.githubusercontent.com/46249224/88269032-019a9200-ccdc-11ea-8a23-5501c99d4f01.png)